util/dvsim/Launcher.py

```diff
@@ -263,9 +263,14 @@ def _get_n_lines(pos, num):
                 if chk_failed:
                     if _find_patterns(self.deploy.fail_patterns, line):
                         # If failed, then nothing else to do. Just return.
+                        # For build failures show 10 extra lines in context.
+                        context = (
+                            lines[cnt:cnt+10] if self.deploy.target == "build"
+                            else []
+                        )
                         return "F", ErrorMessage(line_number=cnt + 1,
                                                  message=line.strip(),
-                                                 context=[])
+                                                 context=context)
```

Comment:
```suggestion
                                                 context=lines[-10:])
```

I think it would be better to set the context unconditionally to the last 10 lines. Coverage merge or report for example could exit if `urg` is not accessible, or if those command were buggy. Simulations runs as well, if we made a mistake and the simv is not at the right path (especially on edacloud). These lines will provide more context in those cases, rather than just plainly say "Job returned non-zero exit code".

Reply:
Actually, the old behavior was to print error lines + 5 additional lines. So `context=_get_n_lines(cnt, 5)`

Reply:
I realize now this is for a detected fail pattern, which can be on multiple lines too. You need the context for a detected failure pattern as well. For example, if the bucket is 
```
Error-NOA: Null object access
```

The following 5 lines are useful to know whch object is it talking about.

Reply:
IMO 5 extra following lines for all targets (build, run etc) has remained sufficient so far for all detected failures. It will simplify this implementation. We print the last 10 lines of the log only on unknown failures (the job returning non-zero exit code and all pass patterns not seen cases). I EDITed the suggestion in my first comment. My apologies for the 5 vs 10 line confusion. 

Also, could you remove the `_get_n_lines` function above? It seems unused.

Reply:
I think this version does as expected. The _get_n_lines function is gone.